src/Agent.Worker/Build/BuildJobExtension.cs

```diff
@@ -223,6 +222,16 @@ private bool TrySetPrimaryEndpointAndProviderInfo(IExecutionContext executionCon
                 if (SourceProvider != null)
                 {
                     SourceEndpoint = ep;
+
+                    // set endpoint id to repository id
+                    string repoIdString;
+                    Guid repoId;
+                    if (SourceEndpoint.Data.TryGetValue("repositoryId", out repoIdString) &&
+                        Guid.TryParse(repoIdString, out repoId))
```

Comment:
:D totally forgot TFVC. 
is there any change endpoint ID still matter?
old agent: endpoint id = repoid or $/
new agent: endpoint id = guid.empty.